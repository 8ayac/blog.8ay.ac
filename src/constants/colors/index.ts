export * from './blue';
export * from './green';
export * from './yellow';
