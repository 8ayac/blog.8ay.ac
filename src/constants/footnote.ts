export const FOOTNOTE_DEFINITION_PREFIX = 'fn:';
export const FOOTNOTE_REFERENCE_PREFIX = 'fnref:';
