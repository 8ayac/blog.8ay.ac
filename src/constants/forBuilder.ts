export const CONTENT_OUTPUT_DIR = 'src/shared/.content';
export const ARTICLES_DIR = 'docs/articles';
