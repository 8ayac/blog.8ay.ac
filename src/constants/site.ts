export const BLOG_TITLE = 'もしくはこれ';
export const BLOG_SUBTITLE = "8ayac's blog";
