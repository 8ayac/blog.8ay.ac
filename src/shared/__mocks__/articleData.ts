export const mockArticleData = {
  t1: {
    // Also as a default
    attributes: {
      id: 'test-article-id-1',
      title: 'Test Article Title 1',
      publishedAt: '2021-01-01T00:00:00.000Z',
      tags: ['test-tag-1-1', 'test-tag-1-2'],
    },
    body:
      '## test h2\r\n\r\nthis is a test paragraph of test-article-id-1.\r\n',
    description:
      'test h2\n\nthis is a test paragraph of test-article-id-1.\n\n',
    changeLogs: [
      {
        id: '1000001',
        date: '2021-01-01T11:11:11.000Z',
        description: ':test_prefix: this is a test commit 1-1',
        author: '8ayac',
      },
      {
        id: '1000002',
        date: '2021-01-01T22:22:22.000Z',
        description: ':test_prefix: this is a test commit 1-2',
        author: '8ayac',
      },
    ],
  },

  t2: {
    attributes: {
      id: 'test-article-id-2',
      title: 'Test Article Title 2',
      publishedAt: '2021-02-01T00:00:00.000Z',
      tags: ['test-tag-2-1', 'test-tag-2-2'],
    },
    body:
      '## test h2\r\n\r\nthis is a test paragraph of test-article-id-2.\r\n',
    description:
      'test h2\n\nthis is a test paragraph of test-article-id-2.\n\n',
    changeLogs: [
      {
        id: '2000001',
        date: '2021-02-01T11:11:11.000Z',
        description: ':test_prefix: this is a test commit 2-1',
        author: '8ayac',
      },
      {
        id: '2000002',
        date: '2021-02-01T22:22:22.000Z',
        description: ':test_prefix: this is a test commit 2-2',
        author: '8ayac',
      },
    ],
  },
};
