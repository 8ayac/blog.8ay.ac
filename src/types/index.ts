export type ArticleAttribute = {
  title: string;
  publishedAt: Date;
  updatedAt: Date;
  tags?: string[];
};

export type Article = ArticleAttribute & {
  body: string;
};
