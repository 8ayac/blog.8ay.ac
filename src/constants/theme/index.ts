import * as colors from '@/src/constants/colors';

export const theme = {
  color: {
    blue: colors.blue,
    green: colors.green,
    yellow: colors.yellow,

    primary: colors.blue.base,
    primaryLight: colors.blue.light,
    primaryDark: colors.blue.dark,

    text: {
      primary: 'hsla(0, 0%, 15%)',
      primaryLight: 'hsla(0, 0%, 40%)',
      secondary: colors.blue.dark2,
      accent: colors.green.dark,
    },
  },
};
