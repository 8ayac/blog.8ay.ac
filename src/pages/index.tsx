import { BodyHeader } from '@/src/components/elements/BodyHeader';
import { ArticleHeaderList } from '@/src/components/widgets/ArticleHeaderList';
import { CategoryTagList } from '@/src/components/widgets/CategoryTagList';
import jsonArticles from '@/src/shared/.content/articles.json';
import { getNumberOfArticlesWithATag } from '@/src/shared/utils';
import { Article } from '@/src/types';
import styled, { StyledComponent } from '@emotion/styled';
import { GetStaticProps, InferGetStaticPropsType, NextPage } from 'next';
import React from 'react';

const S: { [key: string]: StyledComponent<any> } = {
  Section: styled.section`
    margin-bottom: 4rem;
  `,
};
S.ArticleListSection = styled(S.Section)``;
S.CategoryListSection = styled(S.Section)``;

export const getStaticProps: GetStaticProps = async () => {
  return {
    props: {
      articles: jsonArticles,
    },
  };
};

const TopPage: NextPage<InferGetStaticPropsType<typeof getStaticProps>> = ({
  articles,
}) => {
  return (
    <>
      <S.ArticleListSection>
        <BodyHeader title="Articles" />
        <ArticleHeaderList articles={articles} />
      </S.ArticleListSection>

      <S.CategoryListSection>
        <BodyHeader title="Tags" />
        <CategoryTagList
          tags={(articles as Article[])
            .map((article) => article.tags)
            .flat()
            .map(
              (tag) => `${tag} (${getNumberOfArticlesWithATag(tag, articles)})`
            )}
        />
      </S.CategoryListSection>
    </>
  );
};

export default TopPage;
