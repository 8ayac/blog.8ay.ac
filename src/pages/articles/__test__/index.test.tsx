import { theme } from '@/src/constants/theme';
import ArticleIndexPage from '@/src/pages/articles';
import { mockArticleData } from '@/src/shared/__mocks__/articleData';
import { Article } from '@/src/types';
import { ThemeProvider } from '@emotion/react';
import { mount } from 'enzyme';
import React from 'react';

describe('ArticleIndexPage', () => {
  const testArticles: Article[] = [mockArticleData.t1, mockArticleData.t2];

  test('is rendered correctly to match the snapshot', () => {
    const wrapper = mount(
      <ThemeProvider theme={theme}>
        <ArticleIndexPage articles={testArticles} />
      </ThemeProvider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
