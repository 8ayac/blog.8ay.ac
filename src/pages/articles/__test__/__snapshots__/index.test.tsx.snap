// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleIndexPage is rendered correctly to match the snapshot 1`] = `
.emotion-0 {
  margin-bottom: 3rem;
  border-bottom: 0.25rem solid hsla(0, 0%, 75%);
}

.emotion-2 {
  margin: 0 0 1rem;
  font-size: 2.16em;
}

.emotion-4 {
  display: grid;
  grid-gap: 5%;
  margin: 0 auto;
}

.emotion-6 h1 {
  font-size: 2.3rem;
}

.emotion-10 {
  margin-bottom: 0.3rem;
}

.emotion-12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-flow: row wrap;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 1rem;
}

.emotion-12>* {
  margin-right: 1rem;
}

.emotion-14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-flow: row wrap;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
}

.emotion-16 {
  margin-right: 0.5rem;
  font-weight: 600;
}

.emotion-18 {
  font-weight: bolder;
  color: hsla(212, 72%, 40%);
}

.emotion-28 {
  margin: 0;
  color: hsla(0, 0%, 15%);
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-filter: drop-shadow(2px 2px 1px hsla(148, 72%, 80%));
  filter: drop-shadow(2px 2px 1px hsla(148, 72%, 80%));
}

.emotion-30 {
  margin-top: 0.3rem;
}

.emotion-32 {
  margin-top: 2rem;
  margin-bottom: 0.5rem;
}

.emotion-34 {
  margin-right: 1rem;
  margin-left: 0.5rem;
  font-size: 1em;
  color: hsla(0, 0%, 40%);
}

.emotion-36 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin: 0 0.7rem 0.7rem 0;
}

.emotion-38 {
  padding: 3% 1rem 4%;
  margin: 0;
  font-size: 0.8em;
  font-weight: 600;
  color: white;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: hsla(212, 72%, 50%);
  -webkit-filter: drop-shadow(2px 2px 1px hsla(212, 72%, 80%));
  filter: drop-shadow(2px 2px 1px hsla(212, 72%, 80%));
  border-radius: 3px;
}

.emotion-38:hover {
  color: white;
  background-color: hsla(212, 72%, 60%);
}

<ArticleIndexPage
  articles={
    Array [
      Object {
        "body": "bluhbluhbluh",
        "id": "test-article-01",
        "publishedAt": 2021-01-01T00:00:00.000Z,
        "tags": Array [
          "test01",
        ],
        "title": "Test Title 01",
        "updatedAt": 2021-01-01T00:00:00.000Z,
      },
      Object {
        "body": "bluhbluhbluh",
        "id": "test-article-02",
        "publishedAt": 2021-02-02T00:00:00.000Z,
        "tags": Array [
          "test02",
        ],
        "title": "Test Title 02",
        "updatedAt": 2021-02-02T00:00:00.000Z,
      },
    ]
  }
>
  <BodyHeader
    title="Articles"
  >
    <WrapperHeader>
      <header
        className="emotion-0 emotion-1"
      >
        <TitleH1>
          <h1
            className="emotion-2 emotion-3"
          >
            Articles
          </h1>
        </TitleH1>
      </header>
    </WrapperHeader>
  </BodyHeader>
  <ArticleHeaderList
    articles={
      Array [
        Object {
          "body": "bluhbluhbluh",
          "id": "test-article-01",
          "publishedAt": 2021-01-01T00:00:00.000Z,
          "tags": Array [
            "test01",
          ],
          "title": "Test Title 01",
          "updatedAt": 2021-01-01T00:00:00.000Z,
        },
        Object {
          "body": "bluhbluhbluh",
          "id": "test-article-02",
          "publishedAt": 2021-02-02T00:00:00.000Z,
          "tags": Array [
            "test02",
          ],
          "title": "Test Title 02",
          "updatedAt": 2021-02-02T00:00:00.000Z,
        },
      ]
    }
  >
    <ComponentWrapperDiv>
      <div
        className="emotion-4 emotion-5"
      >
        <ArticleHeaderWrapperDiv>
          <div
            className="emotion-6 emotion-7"
          >
            <ArticleHeader
              article={
                Object {
                  "body": "bluhbluhbluh",
                  "id": "test-article-01",
                  "publishedAt": 2021-01-01T00:00:00.000Z,
                  "tags": Array [
                    "test01",
                  ],
                  "title": "Test Title 01",
                  "updatedAt": 2021-01-01T00:00:00.000Z,
                }
              }
              key="test-article-01"
            >
              <BodyHeader
                afterTitle={
                  <TagListWrapperDiv>
                    <FontAwesomeIcon
                      css="unknown styles"
                      icon={
                        Object {
                          "icon": Array [
                            640,
                            512,
                            Array [],
                            "f02c",
                            "M497.941 225.941L286.059 14.059A48 48 0 0 0 252.118 0H48C21.49 0 0 21.49 0 48v204.118a48 48 0 0 0 14.059 33.941l211.882 211.882c18.744 18.745 49.136 18.746 67.882 0l204.118-204.118c18.745-18.745 18.745-49.137 0-67.882zM112 160c-26.51 0-48-21.49-48-48s21.49-48 48-48 48 21.49 48 48-21.49 48-48 48zm513.941 133.823L421.823 497.941c-18.745 18.745-49.137 18.745-67.882 0l-.36-.36L527.64 323.522c16.999-16.999 26.36-39.6 26.36-63.64s-9.362-46.641-26.36-63.64L331.397 0h48.721a48 48 0 0 1 33.941 14.059l211.882 211.882c18.745 18.745 18.745 49.137 0 67.882z",
                          ],
                          "iconName": "tags",
                          "prefix": "fas",
                        }
                      }
                    />
                    <CategoryTagList
                      tags={
                        Array [
                          "test01",
                        ]
                      }
                    />
                  </TagListWrapperDiv>
                }
                beforeTitle={
                  <DateTimeWrapperDiv>
                    <DateTime
                      date={2021-01-01T00:00:00.000Z}
                      description="Published At"
                    />
                    <DateTime
                      date={2021-01-01T00:00:00.000Z}
                      description="Updated At"
                    />
                  </DateTimeWrapperDiv>
                }
                linkTo="/articles/2021-01-01/test-article-01"
                title="Test Title 01"
              >
                <WrapperHeader>
                  <header
                    className="emotion-0 emotion-1"
                  >
                    <BeforeTitleDiv>
                      <div
                        className="emotion-10 emotion-11"
                      >
                        <DateTimeWrapperDiv>
                          <div
                            className="emotion-12 emotion-13"
                          >
                            <DateTime
                              date={2021-01-01T00:00:00.000Z}
                              description="Published At"
                            >
                              <ComponentWrapperDiv>
                                <div
                                  className="emotion-14 emotion-15"
                                >
                                  <DescriptionSpan>
                                    <span
                                      className="emotion-16 emotion-17"
                                    >
                                      Published At:
                                    </span>
                                  </DescriptionSpan>
                                  <DateTime>
                                    <time
                                      className="emotion-18 emotion-19"
                                    >
                                      2021-01-01T00:00:00.000Z
                                    </time>
                                  </DateTime>
                                </div>
                              </ComponentWrapperDiv>
                            </DateTime>
                            <DateTime
                              date={2021-01-01T00:00:00.000Z}
                              description="Updated At"
                            >
                              <ComponentWrapperDiv>
                                <div
                                  className="emotion-14 emotion-15"
                                >
                                  <DescriptionSpan>
                                    <span
                                      className="emotion-16 emotion-17"
                                    >
                                      Updated At:
                                    </span>
                                  </DescriptionSpan>
                                  <DateTime>
                                    <time
                                      className="emotion-18 emotion-19"
                                    >
                                      2021-01-01T00:00:00.000Z
                                    </time>
                                  </DateTime>
                                </div>
                              </ComponentWrapperDiv>
                            </DateTime>
                          </div>
                        </DateTimeWrapperDiv>
                      </div>
                    </BeforeTitleDiv>
                    <TitleH1>
                      <h1
                        className="emotion-2 emotion-3"
                      >
                        <Link
                          href="/articles/2021-01-01/test-article-01"
                          passHref={true}
                        >
                          <TitleAnchor
                            href="/articles/2021-01-01/test-article-01"
                            onClick={[Function]}
                            onMouseEnter={[Function]}
                          >
                            <a
                              className="emotion-28 emotion-29"
                              href="/articles/2021-01-01/test-article-01"
                              onClick={[Function]}
                              onMouseEnter={[Function]}
                            >
                              Test Title 01
                            </a>
                          </TitleAnchor>
                        </Link>
                      </h1>
                    </TitleH1>
                    <AfterTitleDiv>
                      <div
                        className="emotion-30 emotion-31"
                      >
                        <TagListWrapperDiv>
                          <div
                            className="emotion-32 emotion-33"
                          >
                            <FontAwesomeIcon
                              border={false}
                              className="emotion-34"
                              fixedWidth={false}
                              flip={null}
                              icon={
                                Object {
                                  "icon": Array [
                                    640,
                                    512,
                                    Array [],
                                    "f02c",
                                    "M497.941 225.941L286.059 14.059A48 48 0 0 0 252.118 0H48C21.49 0 0 21.49 0 48v204.118a48 48 0 0 0 14.059 33.941l211.882 211.882c18.744 18.745 49.136 18.746 67.882 0l204.118-204.118c18.745-18.745 18.745-49.137 0-67.882zM112 160c-26.51 0-48-21.49-48-48s21.49-48 48-48 48 21.49 48 48-21.49 48-48 48zm513.941 133.823L421.823 497.941c-18.745 18.745-49.137 18.745-67.882 0l-.36-.36L527.64 323.522c16.999-16.999 26.36-39.6 26.36-63.64s-9.362-46.641-26.36-63.64L331.397 0h48.721a48 48 0 0 1 33.941 14.059l211.882 211.882c18.745 18.745 18.745 49.137 0 67.882z",
                                  ],
                                  "iconName": "tags",
                                  "prefix": "fas",
                                }
                              }
                              inverse={false}
                              listItem={false}
                              mask={null}
                              pull={null}
                              pulse={false}
                              rotation={null}
                              size={null}
                              spin={false}
                              swapOpacity={false}
                              symbol={false}
                              title=""
                              transform={null}
                            >
                              <svg
                                aria-hidden="true"
                                className="svg-inline--fa fa-tags fa-w-20 emotion-34"
                                data-icon="tags"
                                data-prefix="fas"
                                focusable="false"
                                role="img"
                                style={Object {}}
                                viewBox="0 0 640 512"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <path
                                  d="M497.941 225.941L286.059 14.059A48 48 0 0 0 252.118 0H48C21.49 0 0 21.49 0 48v204.118a48 48 0 0 0 14.059 33.941l211.882 211.882c18.744 18.745 49.136 18.746 67.882 0l204.118-204.118c18.745-18.745 18.745-49.137 0-67.882zM112 160c-26.51 0-48-21.49-48-48s21.49-48 48-48 48 21.49 48 48-21.49 48-48 48zm513.941 133.823L421.823 497.941c-18.745 18.745-49.137 18.745-67.882 0l-.36-.36L527.64 323.522c16.999-16.999 26.36-39.6 26.36-63.64s-9.362-46.641-26.36-63.64L331.397 0h48.721a48 48 0 0 1 33.941 14.059l211.882 211.882c18.745 18.745 18.745 49.137 0 67.882z"
                                  fill="currentColor"
                                  style={Object {}}
                                />
                              </svg>
                            </FontAwesomeIcon>
                            <CategoryTagList
                              tags={
                                Array [
                                  "test01",
                                ]
                              }
                            >
                              <ComponentWrapperDiv
                                key="test01"
                              >
                                <div
                                  className="emotion-36 emotion-37"
                                >
                                  <CategoryTag
                                    name="test01"
                                  >
                                    <Link
                                      href="/tags/test01"
                                      passHref={true}
                                    >
                                      <Anchor
                                        href="/tags/test01"
                                        onClick={[Function]}
                                        onMouseEnter={[Function]}
                                      >
                                        <a
                                          className="emotion-38 emotion-39"
                                          href="/tags/test01"
                                          onClick={[Function]}
                                          onMouseEnter={[Function]}
                                        >
                                          test01
                                        </a>
                                      </Anchor>
                                    </Link>
                                  </CategoryTag>
                                </div>
                              </ComponentWrapperDiv>
                            </CategoryTagList>
                          </div>
                        </TagListWrapperDiv>
                      </div>
                    </AfterTitleDiv>
                  </header>
                </WrapperHeader>
              </BodyHeader>
            </ArticleHeader>
            <ArticleHeader
              article={
                Object {
                  "body": "bluhbluhbluh",
                  "id": "test-article-02",
                  "publishedAt": 2021-02-02T00:00:00.000Z,
                  "tags": Array [
                    "test02",
                  ],
                  "title": "Test Title 02",
                  "updatedAt": 2021-02-02T00:00:00.000Z,
                }
              }
              key="test-article-02"
            >
              <BodyHeader
                afterTitle={
                  <TagListWrapperDiv>
                    <FontAwesomeIcon
                      css="unknown styles"
                      icon={
                        Object {
                          "icon": Array [
                            640,
                            512,
                            Array [],
                            "f02c",
                            "M497.941 225.941L286.059 14.059A48 48 0 0 0 252.118 0H48C21.49 0 0 21.49 0 48v204.118a48 48 0 0 0 14.059 33.941l211.882 211.882c18.744 18.745 49.136 18.746 67.882 0l204.118-204.118c18.745-18.745 18.745-49.137 0-67.882zM112 160c-26.51 0-48-21.49-48-48s21.49-48 48-48 48 21.49 48 48-21.49 48-48 48zm513.941 133.823L421.823 497.941c-18.745 18.745-49.137 18.745-67.882 0l-.36-.36L527.64 323.522c16.999-16.999 26.36-39.6 26.36-63.64s-9.362-46.641-26.36-63.64L331.397 0h48.721a48 48 0 0 1 33.941 14.059l211.882 211.882c18.745 18.745 18.745 49.137 0 67.882z",
                          ],
                          "iconName": "tags",
                          "prefix": "fas",
                        }
                      }
                    />
                    <CategoryTagList
                      tags={
                        Array [
                          "test02",
                        ]
                      }
                    />
                  </TagListWrapperDiv>
                }
                beforeTitle={
                  <DateTimeWrapperDiv>
                    <DateTime
                      date={2021-02-02T00:00:00.000Z}
                      description="Published At"
                    />
                    <DateTime
                      date={2021-02-02T00:00:00.000Z}
                      description="Updated At"
                    />
                  </DateTimeWrapperDiv>
                }
                linkTo="/articles/2021-02-02/test-article-02"
                title="Test Title 02"
              >
                <WrapperHeader>
                  <header
                    className="emotion-0 emotion-1"
                  >
                    <BeforeTitleDiv>
                      <div
                        className="emotion-10 emotion-11"
                      >
                        <DateTimeWrapperDiv>
                          <div
                            className="emotion-12 emotion-13"
                          >
                            <DateTime
                              date={2021-02-02T00:00:00.000Z}
                              description="Published At"
                            >
                              <ComponentWrapperDiv>
                                <div
                                  className="emotion-14 emotion-15"
                                >
                                  <DescriptionSpan>
                                    <span
                                      className="emotion-16 emotion-17"
                                    >
                                      Published At:
                                    </span>
                                  </DescriptionSpan>
                                  <DateTime>
                                    <time
                                      className="emotion-18 emotion-19"
                                    >
                                      2021-02-02T00:00:00.000Z
                                    </time>
                                  </DateTime>
                                </div>
                              </ComponentWrapperDiv>
                            </DateTime>
                            <DateTime
                              date={2021-02-02T00:00:00.000Z}
                              description="Updated At"
                            >
                              <ComponentWrapperDiv>
                                <div
                                  className="emotion-14 emotion-15"
                                >
                                  <DescriptionSpan>
                                    <span
                                      className="emotion-16 emotion-17"
                                    >
                                      Updated At:
                                    </span>
                                  </DescriptionSpan>
                                  <DateTime>
                                    <time
                                      className="emotion-18 emotion-19"
                                    >
                                      2021-02-02T00:00:00.000Z
                                    </time>
                                  </DateTime>
                                </div>
                              </ComponentWrapperDiv>
                            </DateTime>
                          </div>
                        </DateTimeWrapperDiv>
                      </div>
                    </BeforeTitleDiv>
                    <TitleH1>
                      <h1
                        className="emotion-2 emotion-3"
                      >
                        <Link
                          href="/articles/2021-02-02/test-article-02"
                          passHref={true}
                        >
                          <TitleAnchor
                            href="/articles/2021-02-02/test-article-02"
                            onClick={[Function]}
                            onMouseEnter={[Function]}
                          >
                            <a
                              className="emotion-28 emotion-29"
                              href="/articles/2021-02-02/test-article-02"
                              onClick={[Function]}
                              onMouseEnter={[Function]}
                            >
                              Test Title 02
                            </a>
                          </TitleAnchor>
                        </Link>
                      </h1>
                    </TitleH1>
                    <AfterTitleDiv>
                      <div
                        className="emotion-30 emotion-31"
                      >
                        <TagListWrapperDiv>
                          <div
                            className="emotion-32 emotion-33"
                          >
                            <FontAwesomeIcon
                              border={false}
                              className="emotion-34"
                              fixedWidth={false}
                              flip={null}
                              icon={
                                Object {
                                  "icon": Array [
                                    640,
                                    512,
                                    Array [],
                                    "f02c",
                                    "M497.941 225.941L286.059 14.059A48 48 0 0 0 252.118 0H48C21.49 0 0 21.49 0 48v204.118a48 48 0 0 0 14.059 33.941l211.882 211.882c18.744 18.745 49.136 18.746 67.882 0l204.118-204.118c18.745-18.745 18.745-49.137 0-67.882zM112 160c-26.51 0-48-21.49-48-48s21.49-48 48-48 48 21.49 48 48-21.49 48-48 48zm513.941 133.823L421.823 497.941c-18.745 18.745-49.137 18.745-67.882 0l-.36-.36L527.64 323.522c16.999-16.999 26.36-39.6 26.36-63.64s-9.362-46.641-26.36-63.64L331.397 0h48.721a48 48 0 0 1 33.941 14.059l211.882 211.882c18.745 18.745 18.745 49.137 0 67.882z",
                                  ],
                                  "iconName": "tags",
                                  "prefix": "fas",
                                }
                              }
                              inverse={false}
                              listItem={false}
                              mask={null}
                              pull={null}
                              pulse={false}
                              rotation={null}
                              size={null}
                              spin={false}
                              swapOpacity={false}
                              symbol={false}
                              title=""
                              transform={null}
                            >
                              <svg
                                aria-hidden="true"
                                className="svg-inline--fa fa-tags fa-w-20 emotion-34"
                                data-icon="tags"
                                data-prefix="fas"
                                focusable="false"
                                role="img"
                                style={Object {}}
                                viewBox="0 0 640 512"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <path
                                  d="M497.941 225.941L286.059 14.059A48 48 0 0 0 252.118 0H48C21.49 0 0 21.49 0 48v204.118a48 48 0 0 0 14.059 33.941l211.882 211.882c18.744 18.745 49.136 18.746 67.882 0l204.118-204.118c18.745-18.745 18.745-49.137 0-67.882zM112 160c-26.51 0-48-21.49-48-48s21.49-48 48-48 48 21.49 48 48-21.49 48-48 48zm513.941 133.823L421.823 497.941c-18.745 18.745-49.137 18.745-67.882 0l-.36-.36L527.64 323.522c16.999-16.999 26.36-39.6 26.36-63.64s-9.362-46.641-26.36-63.64L331.397 0h48.721a48 48 0 0 1 33.941 14.059l211.882 211.882c18.745 18.745 18.745 49.137 0 67.882z"
                                  fill="currentColor"
                                  style={Object {}}
                                />
                              </svg>
                            </FontAwesomeIcon>
                            <CategoryTagList
                              tags={
                                Array [
                                  "test02",
                                ]
                              }
                            >
                              <ComponentWrapperDiv
                                key="test02"
                              >
                                <div
                                  className="emotion-36 emotion-37"
                                >
                                  <CategoryTag
                                    name="test02"
                                  >
                                    <Link
                                      href="/tags/test02"
                                      passHref={true}
                                    >
                                      <Anchor
                                        href="/tags/test02"
                                        onClick={[Function]}
                                        onMouseEnter={[Function]}
                                      >
                                        <a
                                          className="emotion-38 emotion-39"
                                          href="/tags/test02"
                                          onClick={[Function]}
                                          onMouseEnter={[Function]}
                                        >
                                          test02
                                        </a>
                                      </Anchor>
                                    </Link>
                                  </CategoryTag>
                                </div>
                              </ComponentWrapperDiv>
                            </CategoryTagList>
                          </div>
                        </TagListWrapperDiv>
                      </div>
                    </AfterTitleDiv>
                  </header>
                </WrapperHeader>
              </BodyHeader>
            </ArticleHeader>
          </div>
        </ArticleHeaderWrapperDiv>
      </div>
    </ComponentWrapperDiv>
  </ArticleHeaderList>
</ArticleIndexPage>
`;
